common/src/main/java/com/sequenceiq/cloudbreak/ccm/endpoint/ServiceFamily.java

```diff
@@ -0,0 +1,60 @@
+package com.sequenceiq.cloudbreak.ccm.endpoint;
+
+import java.util.Optional;
+
+import javax.annotation.Nonnull;
+
+/**
+ * Represents a family of services.
+ *
+ * @param <T> the type of service endpoint used by the family
+ */
+public interface ServiceFamily<T extends ServiceEndpoint> {
+
+    /**
+     * The service family for nginx web servers on gateways.
+     */
+    ServiceFamily<HttpsServiceEndpoint> GATEWAY =
+            new HttpsServiceFamily(9443, KnownServiceIdentifier.GATEWAY);
+
+    /**
+     * The service family for Apache Knox proxy servers.
+     */
+    ServiceFamily<HttpsServiceEndpoint> KNOX =
+            new HttpsServiceFamily(8443, KnownServiceIdentifier.KNOX);
+
+    /**
+     * Returns a custom HTTPS service family.
+     *
+     * @param defaultPort the default port
+     * @return the custom service family
+     */
+    static ServiceFamily<HttpsServiceEndpoint> customHttpsServiceFamily(int defaultPort) {
```

Comment:
I don't see value in this method over just using the HttpsServiceFamily constructor directly. It also doesn't fit in this class because it's specific to https.

Reply:
Changed.